modules/data.atmosphere/R/download.NEONmet.R

```diff
@@ -0,0 +1,486 @@
+        logger.warn("No data available in year ",current_year)
+        next()
+      }
+      startMon <- min(monthsNeeded)
+      endMon <- max(monthsNeeded)
+  
+  #Set up netcdf file, dimensions, and sequence of dates
+    new.file <- all_files[y_idx]
+    if (file.exists(new.file) && !overwrite) {
+      logger.debug("File '", new.file, "' already exists, skipping.")
+      next()
+    }
+    
+    seqTime <- seq(start_date,end_date,by=1800) 
+    datetime <- as.POSIXct(seqTime)
+    results$startdate[y_idx] <- as.character(datetime[1])
+    results$enddate[y_idx] <- as.character(datetime[length(datetime)])
+    
+    days_since_1700 <- datetime - lubridate::ymd_hm("1700-01-01 00:00 UTC")
+    t <- ncdf4::ncdim_def("time", "days since 1700-01-01", as.numeric(days_since_1700))  #define netCDF dimensions for variables
+    timestep <- 1800
+  
+  ## create lat lon dimensions
+    x <- ncdf4::ncdim_def("longitude", "degrees_east", lon)  # define netCDF dimensions for variables
+    y <- ncdf4::ncdim_def("latitude", "degrees_north", lat)
+    xytdim <- list(x, y, t)
+    FillValue = NA
+  
+  #STEPS: Download all months in startdate to enddate for given variable
+  # Read CSV, copy to array, add to NetCDF file
+
+  #TEMPERATURE NEON.DP1.00002 Air Temp profile or NEON.DP1.00003 Triple-aspirated T (preferred)
+    airTempLoc = grep("DP1.00002",availProducts)
+    airTemp3Loc = grep("DP1.00003",availProducts)
+    if (length(airTempLoc)!=0) { 
+      airTempDates = unlist(availDates[airTempLoc[1]])
+      airTempGoodDates = which((airTempDates >= startMon) & (airTempDates <= endMon))
+      nairTemp = length(airTempGoodDates)
+      } else { nairTemp = 0 }
+    if (length(airTemp3Loc)!=0) { 
+      airTemp3Dates = unlist(availDates[airTemp3Loc[1]])
+      airTemp3GoodDates = which((airTemp3Dates >= startMon) & (airTemp3Dates <= endMon))
+      nairTemp3 = length(airTemp3GoodDates)
+      } else { nairTemp3 = 0 }
+    if ((length(airTempLoc)==0) && (length(airTemp3Loc)==0)) {
+      logger.error("Air temperature DP1.00002 or DP1.00003 not available") }
+    if ((nairTemp==0) && (nairTemp3==0)) {
+      logger.error("Air temperature DP1.00002 or DP1.00003 not available in date range ",startMon," ",endMon)
+    }
+  
+  #define NetCDF variable and create NetCDF file
+    airT.var <- ncvar_def(name = "air_temperature", units = "K", dim = xytdim)
+    nc <- ncdf4::nc_create(new.file, vars = airT.var)  #create netCDF file
+    ncdata <- rep(FillValue,length(datetime))
+    if (nairTemp3>nairTemp) {
+      for (mon in airTemp3Dates[airTemp3GoodDates]) {
+        neonData <- nneo::nneo_data(product_code = availProducts[airTemp3Loc[1]], site_code = site, year_month = mon)
+        if (length(neonData$data$urls)>0) {
+          #Extract and read 30 minute data from the highest vertical level among files returned
+          url30 <- tail(sort(neonData$data$urls[grep("*.00000.000.*_30min",neonData$data$urls)]),1)
```

Comment:
Thanks. Note that the grep statement does subtly change across products, but you're right, it's common enough to most to be abstracted into a string.